pkg/provider/aws/resources/lambda.go

```diff
@@ -103,6 +103,44 @@ func (lambda *LambdaFunction) MakeOperational(dag *core.ResourceGraph, appName s
 			return fmt.Errorf("lambda %s has more than one role downstream", lambda.Id())
 		}
 	}
+
+	downstreamVpcs := core.GetDownstreamResourcesOfType[*Vpc](dag, lambda)
+	upstreamVpcs := core.GetUpstreamResourcesOfType[*Vpc](dag, lambda)
+	if len(downstreamVpcs) > 1 {
+		return fmt.Errorf("lambda %s has more than one vpc downstream", lambda.Id())
+	}
+	if len(upstreamVpcs) > 0 && len(downstreamVpcs) == 1 {
+		return fmt.Errorf("lambda %s has an upstream vpc upstream and cannot be associated with a downstream vpc", lambda.Id())
+	}
```

Comment:
i dont think we need to care about upstream vpcs here, not sure why we would have them but they shouldnt quite matter